langchain/utilities/arxiv.py

```diff
@@ -71,21 +72,21 @@ def run(self, query: str) -> str:
         It uses only the most informative fields of article meta information.
         """
         try:
-            docs = [
-                f"Published: {result.updated.date()}\nTitle: {result.title}\n"
-                f"Authors: {', '.join(a.name for a in result.authors)}\n"
-                f"Summary: {result.summary}"
-                for result in self.arxiv_search(  # type: ignore
-                    query[: self.ARXIV_MAX_QUERY_LENGTH], max_results=self.top_k_results
-                ).results()
-            ]
-            return (
-                "\n\n".join(docs)[: self.doc_content_chars_max]
-                if docs
-                else "No good Arxiv Result was found"
-            )
+            results = self.arxiv_search(  # type: ignore
+                query[: self.ARXIV_MAX_QUERY_LENGTH], max_results=self.top_k_results
+            ).results()
         except self.arxiv_exceptions as ex:
             return f"Arxiv exception: {ex}"
+        docs = [
+            f"Published: {result.updated.date()}\nTitle: {result.title}\n"
+            f"Authors: {', '.join(a.name for a in result.authors)}\n"
+            f"Summary: {result.summary}"
+            for result in results
```

Comment:
Why this is not inside try/except block?  result won't be populated if arxiv_search failes..right or it's guranteed that it will return empty list in case of failure.